### PR TITLE
MGMT-1144 Create get versions api

### PR DESCRIPTION
**Pull request by filanov**
Example output:
```
curl -s -X GET "http://192.168.39.173:31340/api/assisted-install/v1/versions" -H "accept: application/json" | jq .
{
  "assisted-installer": "quay.io/ocpmetal/assisted-installer:stable",
  "assisted-installer-service": "quay.io/mfilanov/bm-inventory:test",
  "discovery-agent": "quay.io/ocpmetal/agent:stable",
  "ignition-manifests-and-kubeconfig-generate": "quay.io/ocpmetal/ignition-manifests-and-kubeconfig-generate:stable",
  "image-builder": "quay.io/ocpmetal/installer-image-build:stable"
}
```

### CODE REVIEW

**Comment by avishayt** [comment]
> Example output:
> 
> ```
> curl -s -X GET "http://192.168.39.173:31340/api/assisted-install/v1/versions" -H "accept: application/json" | jq .
> {
>   "assisted-installer": "quay.io/ocpmetal/assisted-installer:stable",
>   "assisted-installer-service": "quay.io/mfilanov/bm-inventory:test",
>   "discovery-agent": "quay.io/ocpmetal/agent:stable",
>   "ignition-manifests-and-kubeconfig-generate": "quay.io/ocpmetal/ignition-manifests-and-kubeconfig-generate:stable",
>   "image-builder": "quay.io/ocpmetal/installer-image-build:stable"
> }
> ```

Is this output meaningful?  "quay.io/ocpmetal/assisted-installer:stable" changes all the time.  Do we want a git hash?

File: swagger.yaml [hunk 1]
```diff
@@ -700,6 +700,18 @@ paths:
           schema:
             $ref: '#/definitions/error'
 
+  /versions:
```
Comment (avishayt) [inline swagger.yaml:703]: Maybe /component_versions or /software_versions?
In the future I'd like to have another API that returns OpenShift versions that the service knows how to install.

Comment (filanov) [reply swagger.yaml:703]: i don't mind changing it to component_versions

**Comment by filanov** [comment]
> > Example output:
> > ```
> > curl -s -X GET "http://192.168.39.173:31340/api/assisted-install/v1/versions" -H "accept: application/json" | jq .
> > {
> >   "assisted-installer": "quay.io/ocpmetal/assisted-installer:stable",
> >   "assisted-installer-service": "quay.io/mfilanov/bm-inventory:test",
> >   "discovery-agent": "quay.io/ocpmetal/agent:stable",
> >   "ignition-manifests-and-kubeconfig-generate": "quay.io/ocpmetal/ignition-manifests-and-kubeconfig-generate:stable",
> >   "image-builder": "quay.io/ocpmetal/installer-image-build:stable"
> > }
> > ```
> 
> Is this output meaningful? "quay.io/ocpmetal/assisted-installer:stable" changes all the time. Do we want a git hash?

The problem is that i don't have this information. if i do docker inspect before i deploy it will be incorrect if a new version will be released on the same tag. 
The idea is that we will have specific versions for example `quay.io/ocpmetal/assisted-installer:v1.0.1` then this information will be relevant, it's a change the @eranco74 is working on.

**Comment by avishayt** [comment]
> > > Example output:
> > > ```
> > > curl -s -X GET "http://192.168.39.173:31340/api/assisted-install/v1/versions" -H "accept: application/json" | jq .
> > > {
> > >   "assisted-installer": "quay.io/ocpmetal/assisted-installer:stable",
> > >   "assisted-installer-service": "quay.io/mfilanov/bm-inventory:test",
> > >   "discovery-agent": "quay.io/ocpmetal/agent:stable",
> > >   "ignition-manifests-and-kubeconfig-generate": "quay.io/ocpmetal/ignition-manifests-and-kubeconfig-generate:stable",
> > >   "image-builder": "quay.io/ocpmetal/installer-image-build:stable"
> > > }
> > > ```
> > 
> > 
> > Is this output meaningful? "quay.io/ocpmetal/assisted-installer:stable" changes all the time. Do we want a git hash?
> 
> The problem is that i don't have this information. if i do docker inspect before i deploy it will be incorrect if a new version will be released on the same tag.
> The idea is that we will have specific versions for example `quay.io/ocpmetal/assisted-installer:v1.0.1` then this information will be relevant, it's a change the @eranco74 is working on.

OK that's good for releases, but still not great for dev.  No way to get git hash?  Like we used to build packages with the git hash as its version?

**Comment by filanov** [comment]
> > > > Example output:
> > > > ```
> > > > curl -s -X GET "http://192.168.39.173:31340/api/assisted-install/v1/versions" -H "accept: application/json" | jq .
> > > > {
> > > >   "assisted-installer": "quay.io/ocpmetal/assisted-installer:stable",
> > > >   "assisted-installer-service": "quay.io/mfilanov/bm-inventory:test",
> > > >   "discovery-agent": "quay.io/ocpmetal/agent:stable",
> > > >   "ignition-manifests-and-kubeconfig-generate": "quay.io/ocpmetal/ignition-manifests-and-kubeconfig-generate:stable",
> > > >   "image-builder": "quay.io/ocpmetal/installer-image-build:stable"
> > > > }
> > > > ```
> > > 
> > > 
> > > Is this output meaningful? "quay.io/ocpmetal/assisted-installer:stable" changes all the time. Do we want a git hash?
> > 
> > 
> > The problem is that i don't have this information. if i do docker inspect before i deploy it will be incorrect if a new version will be released on the same tag.
> > The idea is that we will have specific versions for example `quay.io/ocpmetal/assisted-installer:v1.0.1` then this information will be relevant, it's a change the @eranco74 is working on.
> 
> OK that's good for releases, but still not great for dev. No way to get git hash? Like we used to build packages with the git hash as its version?

If someone will deploy specific version with specific hash then we can get it.

**Comment by filanov** [comment]
Issues with this API.

1. It will show only the current state - If we created cluster with v1 and updated to v2 then we will not know that this cluster was created with v1.
2. If using tag like stable then we will not be able to know the correct version. 
3. Each cluster may be created with a different version.

From the all the issues that i see i think that this API will confuse us more then help us. 
The only case that it will help us is when we have specific tag like v.X.Y.Z
@ronniel1 @romfreiman @eranco74 what do you think?

**Comment by filanov** [comment]
A suggestion that @avishayt had, to add a property to each cluster and host with the versions of the components it was created with.

**Comment by filanov** [comment]
Tested with test-infra - can be reviewed

**Comment by filanov** [comment]
will be pushed with https://github.com/ori-amizur/introspector/pull/23

**Comment by ronniel1** [comment]
> > Example output:
> > ```
> > curl -s -X GET "http://192.168.39.173:31340/api/assisted-install/v1/versions" -H "accept: application/json" | jq .
> > {
> >   "assisted-installer": "quay.io/ocpmetal/assisted-installer:stable",
> >   "assisted-installer-service": "quay.io/mfilanov/bm-inventory:test",
> >   "discovery-agent": "quay.io/ocpmetal/agent:stable",
> >   "ignition-manifests-and-kubeconfig-generate": "quay.io/ocpmetal/ignition-manifests-and-kubeconfig-generate:stable",
> >   "image-builder": "quay.io/ocpmetal/installer-image-build:stable"
> > }
> > ```
> 
> Is this output meaningful? "quay.io/ocpmetal/assisted-installer:stable" changes all the time. Do we want a git hash?

We are planning to start using version tags for the deployments, which will not change over time
Still, maybe we should use the image sha and not tag?
@filanov , @eranco74

**Comment by eranco74** [comment]
> > > Example output:
> > > ```
> > > curl -s -X GET "http://192.168.39.173:31340/api/assisted-install/v1/versions" -H "accept: application/json" | jq .
> > > {
> > >   "assisted-installer": "quay.io/ocpmetal/assisted-installer:stable",
> > >   "assisted-installer-service": "quay.io/mfilanov/bm-inventory:test",
> > >   "discovery-agent": "quay.io/ocpmetal/agent:stable",
> > >   "ignition-manifests-and-kubeconfig-generate": "quay.io/ocpmetal/ignition-manifests-and-kubeconfig-generate:stable",
> > >   "image-builder": "quay.io/ocpmetal/installer-image-build:stable"
> > > }
> > > ```
> > 
> > 
> > Is this output meaningful? "quay.io/ocpmetal/assisted-installer:stable" changes all the time. Do we want a git hash?
> 
> We are planning to start using version tags for the deployments, which will not change over time
> Still, maybe we should use the image sha and not tag?
> @filanov , @eranco74

We are building both, just a matter of decision when we deploy.
(We will need to update the deploy-tag logic - currently we set same tag for all images).

**Comment by filanov** [comment]
> > > > Example output:
> > > > ```
> > > > curl -s -X GET "http://192.168.39.173:31340/api/assisted-install/v1/versions" -H "accept: application/json" | jq .
> > > > {
> > > >   "assisted-installer": "quay.io/ocpmetal/assisted-installer:stable",
> > > >   "assisted-installer-service": "quay.io/mfilanov/bm-inventory:test",
> > > >   "discovery-agent": "quay.io/ocpmetal/agent:stable",
> > > >   "ignition-manifests-and-kubeconfig-generate": "quay.io/ocpmetal/ignition-manifests-and-kubeconfig-generate:stable",
> > > >   "image-builder": "quay.io/ocpmetal/installer-image-build:stable"
> > > > }
> > > > ```
> > > 
> > > 
> > > Is this output meaningful? "quay.io/ocpmetal/assisted-installer:stable" changes all the time. Do we want a git hash?
> > 
> > 
> > We are planning to start using version tags for the deployments, which will not change over time
> > Still, maybe we should use the image sha and not tag?
> > @filanov , @eranco74
> 
> We are building both, just a matter of decision when we deploy.
> (We will need to update the deploy-tag logic - currently we set same tag for all images).

In a real deployment we will probably use only official release with tags. so maybe if i see that i have quay.io/ocpmetal then i can take only the tag. 
On other deployments i will show the full image
How about that?

**Pull request merged**